File: src/main/java/com/example/repository/mongo/TransformerRepository.java
package com.example.repository.mongo;

import com.example.entity.mongo.Transformer;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface TransformerRepository extends MongoRepository<Transformer, Long> {
    Optional<Transformer> findTopByOrderByIdDesc();
}

File: server-new/src/main/kotlin/com/example/repository/TransformerRepository.kt
package com.example.repository

import com.example.items.Transformer
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

@Repository
interface TransformerRepository : MongoRepository<Transformer, Long> {
    fun findByManufacturer(manufacturer: String): List<Transformer>
}